sdk/metric/aggregator/histogram/histogram.go

```diff
@@ -154,11 +174,20 @@ func (c *Aggregator) Merge(oa export.Aggregator, desc *export.Descriptor) error
 		return aggregator.NewInconsistentMergeError(c, oa)
 	}
 
-	c.checkpoint.sum.AddNumber(desc.NumberKind(), o.checkpoint.sum)
-	c.checkpoint.count.AddNumber(core.Uint64NumberKind, o.checkpoint.count)
+	// Lock() synchronize Merge() and Checkpoint() to make sure all operations of
+	// Merge() is done to the same state.
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	current := c.checkpoint()
+	// We assume that the aggregator being merged is not being updated nor checkpoint or this could be inconsistent.
+	ocheckpoint := o.checkpoint()
+
+	current.sum.AddNumber(desc.NumberKind(), ocheckpoint.sum)
+	current.count.AddNumber(core.Uint64NumberKind, ocheckpoint.count)
 
-	for i := 0; i < len(c.current.buckets.Counts); i++ {
-		c.checkpoint.buckets.Counts[i].AddNumber(core.Uint64NumberKind, o.checkpoint.buckets.Counts[i])
+	for i := 0; i < len(current.buckets.Counts); i++ {
```

Comment:
Maybe a dumb question, but doesn't this assume the bucket sizing is going to be the same for both Histograms? Is that a valid assumption?

Reply:
I would say this is assumption is fine, buckets of a histogram should not change with time and this merges histogram of the same metric. I've added a doc that this merges two histograms with the same buckets.